numpy/core/function_base.py

```diff
@@ -1,11 +1,25 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
 
+def _index_deprecate(i, stacklevel=2):
+    try:
+        i = operator.index(i)
+    except TypeError:
+        msg = "%f cannot be safely interpreted as an integer." % i
```

Comment:
Agreed with @seberg -- we still have a responsibility to provide sensible error messages for basic mistakes like `np.linspace(0, 1, np.float32)` (oops, user forgot to write `dtype=`).